tests/integration/test_prometheus_scrape_multiunit.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This test module tests the prometheus_scrape interface with multiple-to-multiple units related.
+
+This test scaling up/down both sides of the relation, and upgrading.
+
+1. Deploy several units of prometheus and several units of a "provider" charm, and relate them.
+2. Confirm all units of prometheus have the correct and same targets and rules.
+3. Upgrade prometheus.
+4. Scale prometheus up and down.
+5. Scale the "provider" charm up and down.
+"""
+
+import asyncio
+import logging
+from collections import namedtuple
+import pytest
+from deepdiff import DeepDiff
+from helpers import (
+    check_prometheus_is_ready,
+    get_head_stats,
+    get_prometheus_active_targets,
+    oci_image,
+)
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+prometheus_app_name = "prometheus"
+prometheus_resources = {"prometheus-image": oci_image("./metadata.yaml", "prometheus-image")}
+tester_app_name = "tester"
+tester_resources = {
+    "prometheus-tester-image": oci_image(
+        "./tests/integration/prometheus-tester/metadata.yaml",
+        "prometheus-tester-image",
+    )
+}
+num_units = 2  # Using the same number of units for both prometheus and the tester
+
+# The period of time required to be idle before `wait_for_idle` returns is set to 90 sec because
+# unit upgrades were observed to take place 40-70 seconds apart.
+idle_period = 90
+
+
+async def test_setup_env(ops_test: OpsTest):
+    await ops_test.model.set_config(
+        {"logging-config": "<root>=WARNING; unit=DEBUG", "update-status-hook-interval": "60m"}
```

Comment:
Why 60m?

Reply:
So apparently update-status was crucial for robustness
```
  tester/0 [idle] blocked: Waiting for Pebble ready
  tester/1 [idle] blocked: Waiting for Pebble ready
```